Several unit tests fail on windows dev env

### Current Behavior

It has been a long time since I have ran the entire suite locally (usually opting to do targeted `describe.only` tests locally and depend on the `ci` for full pass).

However, while developing a new feature, I elected to run the entire suite locally and I noted a few unexpected failures. Switiching to `main` branch to verify if I caused them or not, I note the tests failed even without my work added (see below)

NOTE: These are likely just non cross-platform-ready test issues rather than OS operational differences but they will need to be assessed case-by-case.


```
1) _spec.js
       is checking if all .js files have a corresponding _spec.js test file.:
     Error: Missing/empty _spec files:
        test\unit\node-red\settings_spec.js
        test\unit\@node-red\nodes\index_spec.js
        test\unit\@node-red\nodes\core\storage\23-watch_spec.js
        >> SNIPPED FOR BREVITY <<

2) nodes/registry/installer
       installs module
         succeeds when file path is valid node-red module:
     AssertionError: expected 'C:\\Users\\user\\repos\\node-red\\node_modules\\npm\\bin\\npm-cli.js' to match /npm\/bin\/npm-cli\.js$/
      at Assertion.fail (node_modules\should\cjs\should.js:275:17)
      at Assertion.value [as match] (node_modules\should\cjs\should.js:356:19)
      at C:\Users\user\repos\node-red\test\unit\@node-red\registry\lib\installer_spec.js:279:40

  3) exec node
       calling exec
         should exec a simple command with appended value from message:

      AssertionError: expected 'bar more\r\n' to be 'bar more\n'
      + expected - actual

      -bar more
      +bar more
      
      at Assertion.fail (node_modules\should\cjs\should.js:275:17)
      at Assertion.value (node_modules\should\cjs\should.js:356:19)
      at Node._inputCallback (test\nodes\core\function\90-exec_spec.js:137:44)
      at         >> SNIPPED FOR BREVITY <<

  4) exec node
       calling spawn
         should return an error for a bad command:
     AssertionError: expected 1 to be below 0
      at Assertion.fail (node_modules\should\cjs\should.js:275:17)
      at Assertion.value (node_modules\should\cjs\should.js:356:19)
      at Node._inputCallback (test\nodes\core\function\90-exec_spec.js:713:52)
      at         >> SNIPPED FOR BREVITY <<

  5) exec node
       calling spawn
         should preserve existing properties on msg object for a failing command:
     Error: Timeout of 3000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\Users\user\repos\node-red\test\nodes\core\function\90-exec_spec.js)
      at listOnTimeout (node:internal/timers:588:17)
      at process.processTimers (node:internal/timers:523:7)

  6) file Nodes
       file out Node
         should write to a file using JSONata:
     Error: Timeout of 3000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\Users\user\repos\node-red\test\nodes\core\storage\10-file_spec.js)
      at listOnTimeout (node:internal/timers:588:17)
      at process.processTimers (node:internal/timers:523:7)

  7) file Nodes
       file out Node
         should append to a file and add newline, except last line of multipart input:
     AssertionError: expected 'Line1\r\nLine2\r\nLine3\r\nLine4' to have property length of 23 (got 26)
      at Assertion.fail (node_modules\should\cjs\should.js:275:17)
      at Assertion.value (node_modules\should\cjs\should.js:356:19)
      at Node._inputCallback (test\nodes\core\storage\10-file_spec.js:243:47)
      at         >> SNIPPED FOR BREVITY <<

  8) file Nodes
       file in Node
         should read in a file using JSONata and output a utf8 string:
     Error: Timeout of 3000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (C:\Users\user\repos\node-red\test\nodes\core\storage\10-file_spec.js)
      at listOnTimeout (node:internal/timers:588:17)
      at process.processTimers (node:internal/timers:523:7)
```

### Expected Behavior

tests should pass regardless of OS

### Steps To Reproduce

run `npm run test` on a windows dev machine

### Example flow

```
paste your flow here
```


### Environment

- Node-RED version: `main` (5.0.1)
- Node.js version: 22.22.0
- npm version:
- Platform/OS: Win 11
- Browser:

--- a/test/nodes/core/function/90-exec_spec.js
+++ b/test/nodes/core/function/90-exec_spec.js
@@ -127,6 +127,7 @@ describe('exec node', function() {
         it('should exec a simple command with appended value from message', function (done) {
             var flow = [{id:"n1", type:"exec", wires:[["n2"]], command:"echo", addpay:"topic", append:"more", oldrc:"false"},
                         {id:"n2", type:"helper"}];
+            var expected = (osType === "Windows_NT") ? "bar more\r\n" : "bar more\n";
             helper.load(execNode, flow, function () {
                 var n1 = helper.getNode("n1");
                 var n2 = helper.getNode("n2");
@@ -134,7 +135,7 @@ describe('exec node', function() {
                     try {
                         msg.should.have.property("payload");
                         msg.payload.should.be.a.String();
-                        msg.payload.should.equal("bar more\n");
+                        msg.payload.should.equal(expected);
                         done();
                     } catch(err) {
                         done(err)
@@ -710,7 +711,13 @@ describe('exec node', function() {
                     try {
                         msg.should.have.property("payload");
                         msg.payload.should.have.property("code");
-                        msg.payload.code.should.be.below(0);
+                        if (osType === "Windows_NT") {
+                            // On Windows the command runs via the shell (cmd.exe), which
+                            // reports an unknown command as a positive exit code (1).
+                            msg.payload.code.should.be.above(0);
+                        } else {
+                            msg.payload.code.should.be.below(0);
+                        }
                         done();
                     }
                     catch(err) { done(err); }
@@ -933,12 +940,15 @@ describe('exec node', function() {
                 var n3 = helper.getNode("n3");
                 var n4 = helper.getNode("n4");
                 var received = 0;
+                var finished = false;
                 var messages = [null,null];
                 var completeTest = function() {
                     if (messages[0] === null || messages[1] === null) {
                         // We have not yet had responses on both ports.
                         return
                     }
+                    if (finished) { return; }
+                    finished = true;
                     try {
                         var msg = messages[0];
                         msg.should.have.property("payload");
@@ -957,6 +967,12 @@ describe('exec node', function() {
                     }
                 };
 
+                // The failing command writes its diagnostic to stderr on Linux
+                // (mkdir) but to stdout on Windows (ping), so listen on both ports.
+                n2.on("input", function(msg) {
+                    messages[0] = msg;
+                    completeTest();
+                });
                 n3.on("input", function(msg) {
                     messages[0] = msg;
                     completeTest();

--- a/test/nodes/core/storage/10-file_spec.js
+++ b/test/nodes/core/storage/10-file_spec.js
@@ -99,7 +99,8 @@ describe('file Nodes', function() {
         });
 
         it('should write to a file using JSONata', function(done) {
-            var fileToTest4jsonata = "'" + resourcesDir + "/'&(20+30)&'-file-test-file.txt'";
+            // Use forward slashes: backslashes in a Windows path would be treated as escape sequences in a JSONata string literal
+            var fileToTest4jsonata = "'" + resourcesDir.replace(/\\/g,"/") + "/'&(20+30)&'-file-test-file.txt'";
             var flow = [{id:"fileNode1", type:"file", name: "fileNode", "filename": fileToTest4jsonata, "filenameType": "jsonata", "appendNewline":false, "overwriteFile":true, wires: [["helperNode1"]]},
                         {id:"helperNode1", type:"helper"}];
             helper.load(fileNode, flow, function() {
@@ -240,7 +241,7 @@ describe('file Nodes', function() {
                                 f.should.equal("Line1\nLine2\nLine3\nLine4");
                             }
                             else {
-                                f.should.have.length(23);
+                                f.should.have.length(26);
                                 f.should.equal("Line1\r\nLine2\r\nLine3\r\nLine4");
                             }
                             done();
@@ -1285,7 +1286,8 @@ describe('file Nodes', function() {
         });
 
         it('should read in a file using JSONata and output a utf8 string', function(done) {
-            var fileToTest4jsonata = "'" + resourcesDir + "/'&(20+30)&'-file-test-file.txt'";
+            // Use forward slashes: backslashes in a Windows path would be treated as escape sequences in a JSONata string literal
+            var fileToTest4jsonata = "'" + resourcesDir.replace(/\\/g,"/") + "/'&(20+30)&'-file-test-file.txt'";
             var flow = [{id:"fileInNode1", type:"file in", name: "fileInNode", "filename":fileToTest4jsonata, "filenameType": "jsonata", "format":"utf8", wires:[["n2"]]},
                         {id:"n2", type:"helper"}];
             helper.load(fileNode, flow, function() {

--- a/test/unit/@node-red/registry/lib/installer_spec.js
+++ b/test/unit/@node-red/registry/lib/installer_spec.js
@@ -276,7 +276,7 @@ describe('nodes/registry/installer', function() {
 
             installer.installModule("foo",null,"/example path/foo-0.1.1.tgz").then(function(info) {
                 const lastCallArgs = exec.run.lastCall.args[1];
-                lastCallArgs[0].should.match(/npm\/bin\/npm-cli\.js$/)
+                lastCallArgs[0].should.match(/npm[\\/]bin[\\/]npm-cli\.js$/)
                 const remainingArgs = lastCallArgs.slice(1);
                 remainingArgs.should.eql([ 'install', '--no-audit', '--no-update-notifier', '--no-fund', '--save', '--save-prefix=~', '--omit=dev', '--engine-strict', '--', '/example path/foo-0.1.1.tgz' ]);
                 info.should.eql(nodeInfo);

--- a/test/unit/_spec.js
+++ b/test/unit/_spec.js
@@ -40,8 +40,10 @@ var walkDirectory = function(dir) {
         var promises = [];
         list.forEach(function(file) {
             var filePath = path.join(dir,file);
+            // Normalise separators so the exclusion checks below work on Windows (\) as well as POSIX (/)
+            var normalisedPath = filePath.split(path.sep).join("/");
 
-            if (!/@node-red\/(editor-client|nodes)/.test(filePath) && !/node-red\/settings\.js/.test(filePath) && !/\/docs\//.test(filePath)) {
+            if (!/@node-red\/(editor-client|nodes)/.test(normalisedPath) && !/node-red\/settings\.js/.test(normalisedPath) && !/\/docs\//.test(normalisedPath)) {
                 promises.push(fs.stat(filePath).then(function(stat){
                     if (stat.isDirectory()) {
                         return walkDirectory(filePath).then(function(results) {

